src/routes/Root/Root.tsx

```diff
@@ -57,6 +60,13 @@ const Root = () => {
   const getActiveClassNames = ({ isActive }: { isActive: boolean }) =>
     isActive ? 'bg-secondary-1-50 w-full rounded-sm py-3 pl-5 -ml-5' : '';
   const name = `${data?.etunimi} ${data?.sukunimi}`;
+  const navigate = useNavigate();
+
+  const changeLanguage = async (lang: LangCode) => {
+    setLangMenuOpen(false);
+    await i18n.changeLanguage(lang);
+    navigate(`/${i18n.language}`);
```

Comment:
I think that the current URL should be set here, so it won't always navigate to front page on language change.